lib/music_theory/envelope.rb

```diff
@@ -0,0 +1,42 @@
+class Envelope
+  attr_accessor :duration, :attack, :sustain, :decay, :sample_rate, :total_frames
+
+  def initialize(options = {})
+    @duration = options[:duration]
+    @attack   = options[:attack]    # in seconds   ex: 0.15
+    @sustain  = options[:sustain]   # percentage   ex: 30
+    @decay    = options[:decay]     # percentage   ex: 60
```

Comment:
maybe use a decimal instead of a percentage (e.g. a number between 0 and 1). it's rare to see actual proportions represented as percentages in code.